### ISSUE TITLE
Label transforms ignore include/exclude filters

### ISSUE THREAD

**Issue by Sal-sal-sal** [TorchIO-project/torchio#1508]
### Description

`RemapLabels`, `RemoveLabels`, and `OneHot` modify label maps that should be excluded by the `include` or `exclude` arguments.

### Reproduction

```python
import torch
import torchio as tio

subject = tio.Subject(
    a=tio.LabelMap(torch.ones(1, 2, 2, 2)),
    b=tio.LabelMap(torch.ones(1, 2, 2, 2)),
)

result = tio.RemapLabels({1: 9}, include=["a"])(subject)

print(result.a.data.unique())  # tensor([9.])
print(result.b.data.unique())  # tensor([9.]), expected tensor([1.])
```

The same problem occurs with:

```python
tio.RemapLabels({1: 9}, exclude=["b"])
tio.RemoveLabels([1], include=["a"])
tio.RemoveLabels([1], exclude=["b"])
tio.OneHot(include=["a"])
tio.OneHot(exclude=["b"])
```

### Expected behavior

Only the selected label maps should change.
Excluded label maps should retain their original values and shape.

### Possible cause

These transforms iterate over `batch.images.items()` directly instead of using `self._get_images(batch)`, which applies the include/exclude filters.

### Version

Reproduced on `main`, commit `2b019d2a9659c838408c73ae77327ac5eb87b190`, TorchIO `2.0.0a2`, using synthetic CPU tensors.

**Comment by romainVala** [comment]
I confirm the bug, and I am surprise that such a fundamental torchio functionality is missing in this v2 version.
@fepegar how copilot miss this point ?

**Comment by fepegar** [comment]
Thanks @Sal-sal-sal for reporting and @romainVala for verifying.

> @fepegar how copilot miss this point ?

AI is unreliable :D

@allcontributors please add @Sal-sal-sal for bug.

**Comment by fepegar** [comment]
@allcontributors please add @Sal-sal-sal for bug.

**Comment by romainVala** [comment]
> > [@fepegar](https://github.com/fepegar) how copilot miss this point ?
> 
> AI is unreliable :D
>

well this is not what we are used to hearing, 
hopfully we will still converge to a robust version 2
